diff --git a/youtube_dl/extractor/twitch.py b/youtube_dl/extractor/twitch.py
index 8c87f6dd3..7be0340f9 100644
--- a/youtube_dl/extractor/twitch.py
+++ b/youtube_dl/extractor/twitch.py
@@ -5,6 +5,7 @@ import itertools
 import re
 import random
 import json
+import subprocess
 
 from .common import InfoExtractor
 from ..compat import (
@@ -337,16 +338,13 @@ class TwitchVodIE(TwitchItemBaseIE):
         if 't' in query:
             info['start_time'] = parse_duration(query['t'][0])
 
-        if info.get('timestamp') is not None:
+        chatlog_proc = subprocess.run(["twitch-chatlog", '%s' % item_id, "-l", "0", "-r"], capture_output=True)
+        if chatlog_proc.returncode == 0:
             info['subtitles'] = {
-                'rechat': [{
-                    'url': update_url_query(
-                        'https://rechat.twitch.tv/rechat-messages', {
-                            'video_id': 'v%s' % item_id,
-                            'start': info['timestamp'],
-                        }),
+                'chat': [{
+                    'data': chatlog_proc.stdout.decode('utf-8'),
                     'ext': 'json',
-                }],
+                }]
             }
 
         return info
